pkg/collector/upgrade/v0_61_0_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upgrade_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
+	"github.com/open-telemetry/opentelemetry-operator/internal/version"
+	"github.com/open-telemetry/opentelemetry-operator/pkg/collector/upgrade"
+)
+
+func Test0_61_0Upgrade(t *testing.T) {
+	collectorInstance := v1alpha1.OpenTelemetryCollector{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "OpenTelemetryCollector",
+			APIVersion: "v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "otel-my-instance",
+			Namespace: "somewhere",
+		},
+		Spec: v1alpha1.OpenTelemetryCollectorSpec{
+			Config: `---
+receivers:
+  jaeger:
+    protocols:
+      grpc:
+    remote_sampling:
+      endpoint: "jaeger-collector:14250"
+      tls:
+        insecure: true
+      strategy_file: "/etc/strategy.json"
+      strategy_file_reload_interval: 10s
+  jaeger/2:
+    protocols:
+      grpc:
+    remote_sampling:
+      endpoint: "jaeger-collector2:14250"
+      tls:
+        insecure: true
+      strategy_file: "/etc/strategy2.json"
+      strategy_file_reload_interval: 15s
+service:
+  pipelines:
+    traces:
+      receivers: [jaeger, jaeger/2]
+      exporters: [nop]
+`,
+		},
+	}
+
+	collectorInstance.Status.Version = "0.60.0"
+	//Test to remove port and change endpoint value.
+	versionUpgrade := &upgrade.VersionUpgrade{
+		Log:      logger,
+		Version:  version.Get(),
+		Client:   k8sClient,
+		Recorder: record.NewFakeRecorder(upgrade.RecordBufferSize),
+	}
+
+	upgradedInstance, err := versionUpgrade.ManagedInstance(context.Background(), collectorInstance)
+	assert.NoError(t, err)
+	assert.Equal(t, `---
+extensions:
+  jaegerremotesampling:
+    source:
+      remote:
+        endpoint: jaeger-collector:14250
+      tls:
+        insecure: true
+      reload_interval: 10s
+      file: /etc/strategy.json
+  jaegerremotesampling:
```

Comment:
Why is the extension `jaegerremotesampling` defined twice? Shouldn't the second one have name e.g. `/2`?